### PR TITLE
Documentation: fix the broken links/dead links

### LINKED ISSUE #10601: CI: Links in docs are not tested
I have attempted fixing some of the most trivial issues with links in our docs (https://github.com/cilium/cilium/pull/10576), but some links are not pass the link checker right now and I'm not sure how to fix these.
Perhaps we could instruct the checker to ignore these somehow? Or explore using an alternative link checker.

Here are some of the main examples of what's broken:

Anchors work fine in the browser, but not in the link checker:

```
(line   35) broken    https://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-quick-configuration - Anchor 'cli-quick-configuration' not found
(line   19) broken    https://github.com/kubernetes-sigs/kubespray#requirements - Anchor 'requirements' not found
```

Commands that contain URL substrings:

```
(line   32) broken    https://github.com/weaveworks/eksctl/releases/download/latest_release/eksctl_$(uname - 404 Client Error: Not Found for url: https://github.com/weaveworks/eksctl/releases/download/latest_release/eksctl_$(uname
(line   39) broken    https://github.com/kubernetes/kops/releases/download/$(curl - 404 Client Error: Not Found for url: https://github.com/kubernetes/kops/releases/download/$(curl
```

This one needs auth:
```
(line   41) broken    https://github.com/cilium/cilium/projects/new - 404 Client Error: Not Found for url: https://github.com/cilium/cilium/projects/new
```

We also have a few localhost links like this:
```
(line   42) broken    http://localhost:9080 - HTTPConnectionPool(host='localhost', port=9080): Max retries exceeded with url: / (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f31cddb0d10>: Failed to establish a new connection: [Errno 111] Connection refused'))
(line  238) broken    http://127.0.0.1:15000/logging?level=debug - HTTPConnectionPool(host='127.0.0.1', port=15000): Max retries exceeded with url: /logging?level=debug (Caused by NewConnectionError('<urllib3.connection.HTTPConnection object at 0x7f31ce248bd0>: Failed to establish a new connection: [Errno 111] Connection refused'))
```

### PR DESCRIPTION
**Pull request by vipul-21**
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [ ] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [ ] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [x] Thanks for contributing!

This the first PR to fix #10601. It addresses the broken link in the documentation as well as the links which can be ignored by the link checker.
The https://github.com/cilium/cilium/issues/10601#issuecomment-1639032222 captures the items we need to close the issue.

Related: #10601

### CODE REVIEW

File: Documentation/network/kubernetes/kubeproxy-free.rst [hunk 1]
```diff
@@ -1633,4 +1633,4 @@ in great details:
       <https://docs.google.com/presentation/d/1VOUcoIxgM_c6M_zAV1dLlRCjyYCMdR3tJv6CEdfLMh8/edit>`__,
       `video <https://fosdem.org/2020/schedule/event/containers_bpf/>`__)
     * "Kernel improvements for Cilium socket LB" (LSF/MM/BPF 2020, `slides
-      <https://docs.google.com/presentation/d/1w2zlpGWV7JUhHYd37El_AUZzyUNSvDfktrF5MJ5G8Bs/edit#slide=id.g746fc02b5b_2_0>`__)
```
Comment (qmonnet) [inline Documentation/network/kubernetes/kubeproxy-free.rst]: It's nice to have the anchors to the right slide, though. I think we can ignore that by configuring the right option for the link checker? I found an old branch in my local repo, where I had this:

```python
linkcheck_anchors_ignore = [
    [...]
    'slide=id.g',
    [...]
]
```

(Actually, aren't these supposed to be skipped given that you have `linkcheck_anchors = False`?)

Same comment for other links to Google presentations.

File: Documentation/contributing/development/hive.rst [hunk 2]
```diff
@@ -903,7 +903,7 @@ test against and allows central control over what data (and at what rate)
 is pulled from the api-server and how it’s stored (in-memory or persisted).
 
 The resources are usually made available centrally for the application,
-e.g. in cilium-agent they’re provided from `pkg/k8s/shared_resources.go <https://github.com/cilium/cilium/blob/main/pkg/k8s/shared_resources.go>`_.
+e.g. in cilium-agent they’re provided from `pkg/k8s/resource.go <https://github.com/cilium/cilium/blob/main/pkg/k8s/resource/resource.go>`_.
```
Comment (qmonnet) [inline Documentation/contributing/development/hive.rst]: Not sure about this one, @pippolo84 could you please confirm this is the right link to use?

File: Documentation/network/kubernetes/compatibility.rst [hunk 3]
```diff
@@ -41,4 +41,4 @@ validation version:
 
 .. include:: compatibility-table.rst
 
-.. _networking.k8s.io/v1: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#networkpolicy-v1-networking-k8s-io
+.. _networking.k8s.io/v1: https://kubernetes.io/docs/concepts/services-networking/network-policies/
```
Comment (qmonnet) [inline Documentation/network/kubernetes/compatibility.rst]: Should this be https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.25/#networkpolicy-v1-networking-k8s-io instead?

File: Documentation/conf.py [hunk 4]
```diff
@@ -261,6 +261,27 @@
...
+    'dns:query',
+
+    # Needs authentication
+    'https://github.com/cilium/cilium/projects/new',
+    'https://github.com/cilium/cilium/settings/branches',
+    'https://jenkins.cilium.io/job/',
+    'https://jenkins.cilium.io/view/'
+]
+
+linkcheck_anchors = False
```
Comment (qmonnet) [inline Documentation/conf.py]: Does this skip checking the anchors? If so, why should we skip them, are they too hard to check?

**Review by qmonnet — CHANGES_REQUESTED** [review]
Looks great overall, thanks!

Please find some links to maybe adjust, and a few questions, inline below.

File: Documentation/network/kubernetes/kubeproxy-free.rst (hunk 1 above)
Comment (vipul-21) [reply Documentation/network/kubernetes/kubeproxy-free.rst]: Yes, they will be ignored by the `linkcheck_anchors`, but when I went to the url the slide id was the id of the first slide. So I removed it from the url.
I can add those back and ignore the ids in `linkcheck_ignore`.

I initially tried to make all the anchors work (the ones that were possible) and added `linkcheck_achor=False` after.

File: Documentation/conf.py (hunk 4 above)
Comment (vipul-21) [reply Documentation/conf.py]: As mentioned above I tried to make work on the anchors and changed the one that were 
broken, but the `md#anchor` even though they exist are throwing the warnings. (not sure why ?)
```
/src/Documentation/gettingstarted/gettinghelp.rst:47: WARNING: broken link: https://github.com/cilium/cilium.io/blob/main/CONTRIBUTING.md#listing-cilium-training (Anchor 'listing-cilium-training' not found)
/src/Documentation/contributing/testing/e2e_legacy.rst:829: WARNING: broken link: https://github.com/cilium/cilium/blob/26dec4c4f4311df2b1a6c909b27ff7fe6e46929f/test/helpers/kubectl.go#L66-L67 (Anchor 'L66-L67' not found)
/src/Documentation/contributing/testing/e2e_legacy.rst:824: WARNING: broken link: https://github.com/cilium/cilium/blob/26dec4c4f4311df2b1a6c909b27ff7fe6e46929f/test/helpers/kubectl.go#L80-L102 (Anchor 'L80-L102' not found)
/src/Documentation/contributing/testing/e2e_legacy.rst:833: WARNING: broken link: https://github.com/cilium/cilium/blob/26dec4c4f4311df2b1a6c909b27ff7fe6e46929f/test/helpers/kubectl.go#L138-L147 (Anchor 'L138-L147' not found)
/src/Documentation/contributing/testing/ci.rst:40: WARNING: broken link: https://github.com/cilium/community/blob/main/CONTRIBUTOR-LADDER.md#organization-member (Anchor 'organization-member' not found)
/src/Documentation/observability/visibility.rst:78: WARNING: broken link: https://github.com/cilium/cilium/tree/master/api/v1/flow#flow-Layer7 (Anchor 'flow-Layer7' not found)
/src/Documentation/contributing/testing/bpf.rst:52: WARNING: broken link: https://github.com/cilium/ebpf/blob/49ebb13083886fc350167f2cde067e094a2b5037/elf_reader.go#L1074 (Anchor 'L1074' not found)
/src/Documentation/contributing/release/backports.rst:314: WARNING: broken link: https://github.com/cilium/release/blob/3c5fc2bdc38f8d290349a612a03cc83655f57a51/pkg/github/labels.go#L26 (Anchor 'L26' not found)
/src/Documentation/security/network/proxy/envoy.rst:17: WARNING: broken link: https://github.com/cilium/proxy#version-compatibility-matrix (Anchor 'version-compatibility-matrix' not found)
/src/Documentation/community/governance/commit_access.rst:210: WARNING: broken link: https://github.com/cncf/foundation/blob/main/README.md#other-content (Anchor 'other-content' not found)

/src/Documentation/network/servicemesh/grpc.rst:61: WARNING: broken link: https://github.com/fullstorydev/grpcurl#binaries (Anchor 'binaries' not found)
/src/Documentation/helm-reference.rst:740: WARNING: broken link: https://github.com/grafana/helm-charts/tree/main/charts/grafana#sidecar-for-dashboards (Anchor 'sidecar-for-dashboards' not found)
/src/Documentation/network/kubernetes/kata.rst:95: WARNING: broken link: https://github.com/kata-containers/packaging/tree/master/kata-deploy#run-a-sample-workload (Anchor 'run-a-sample-workload' not found)
```

File: Documentation/network/kubernetes/kubeproxy-free.rst (hunk 1 above)
Comment (qmonnet) [reply Documentation/network/kubernetes/kubeproxy-free.rst]: OK, please double-check in this case. Maybe some of the anchors point to the first slides, but when I tried the other day, I'm pretty sure at least one of them was pointing to as slide in the middle of the deck. Using `linkcheck_anchors_ignore` seems like a good solution to keep the anchors working; worst case we can ignore them, but I'd rather avoid removing the anchors if we need to point to a particular slide.

File: Documentation/conf.py (hunk 4 above)
Comment (qmonnet) [reply Documentation/conf.py]: I don't know :/. I'm travelling today but can take a look into it next week. Please ping me if I forget.

Comment (qmonnet) [reply Documentation/conf.py]: OK I had a look, I can reproduce but I don't understand either why these anchors are detected as broken. Maybe the link checker doesn't get exactly what we have in a browser.

So OK let's skip anchors for now, checking the rest of the links would already be a nice improvement.

**Comment by qmonnet** [comment]
Could you please:

- Squash the changes from your second commit into the first one
- Split into two commits, but differently: I would like all URL fixes in one, and the `conf.py` additions in a second. Ideally, the changes for `conf.py` should go along enabling the linkchecker in CI, in a separate PR, but I'm OK to have them here in a separate commit.
- Rebase on top of the current `main` branch, this should address the CI issues.

File: Documentation/contributing/development/hive.rst (hunk 2 above)
Comment (pippolo84) [reply Documentation/contributing/development/hive.rst]: The correct link is this one: https://github.com/cilium/cilium/blob/main/daemon/k8s/resources.go

**Comment by vipul-21** [comment]
@qmonnet I think the ideal way makes sense to me as well. I will create another PR with CI enabled and changes to `conf.py` 
This only has the fix for broken links. Let me know if this looks good.

File: Documentation/conf.py [hunk 5]
```diff
@@ -265,7 +265,6 @@
 # Try as hard as possible to find references
 default_role = 'any'
 
-
```
Comment (qmonnet) [inline Documentation/conf.py:268]: Nit: this change is unnecessary.

**Review by qmonnet — APPROVED** [review]
Links update looks all good, thanks!

**Review by pippolo84 — APPROVED** [review]
LGTM, thanks!

**Comment by qmonnet** [comment]
@meyskens @tommyp1ckles Can you please take a quick look at this PR so we can get all reviews in?

**Review by meyskens — APPROVED** [review]
LGTM for me

**Comment by qmonnet** [comment]
@vipul-21: You don't have to rebase the branch every time it gets out-of-date (you only need to do this if there are merge conflicts).

OK, I think this PR is ready. It touches documentation only, all base tests including for docs are green. We don't have a review for `sig-k8s` but we're not changing the contents of the docs, just fixing the links; and the new links all look good to me.

Let's get this merged to fix the docs, and to unblock https://github.com/cilium/cilium/pull/27116.

**Pull request merged**